replica/table.cc

```diff
@@ -508,10 +510,43 @@ void table::enable_off_strategy_trigger() {
 
 std::vector<std::unique_ptr<compaction_group>> table::make_compaction_groups() {
     std::vector<std::unique_ptr<compaction_group>> ret;
-    auto number_of_cg = 1 << _x_log2_compaction_groups;
+    // Avoid shift left 64
+    if (!_x_log2_compaction_groups) {
+        auto&& start_bound = dht::token_range::bound(dht::minimum_token(), true);
+        auto&& end_bound = dht::token_range::bound(dht::maximum_token(), true);
+        ret.emplace_back(std::make_unique<compaction_group>(*this, dht::token_range(std::move(start_bound), std::move(end_bound))));
+        return ret;
+    }
+    int64_t number_of_cg = 1 << _x_log2_compaction_groups;
+    int64_t half_number_of_cg = number_of_cg / 2;
     ret.reserve(number_of_cg);
-    for (auto i = 0; i < number_of_cg; i++) {
-        ret.emplace_back(std::make_unique<compaction_group>(*this));
+    uint8_t log2_shift = (64 - _x_log2_compaction_groups);
+    int64_t key;
+    int64_t keys_per_range = int64_t(1) << log2_shift;
```

Comment:
The token space splitting should be extracted into a helper, it's independent of compaction groups. It would help understandability and could be tested separately. 

Here you would just have:

```
for (auto&& token_range : split_token_range_log2(_x_log2_compaction_groups)) {
    ret.emplace_back(std::make_unique<compaction_group>(*this, token_range));
}
```

Reply:
nice

Reply:
done in v2